src/isMemberOf.ts

```diff
@@ -0,0 +1,41 @@
+import github from '@actions/github';
+import PQueue from 'p-queue';
+import { catchHandler } from './util/catchHandler';
+import { OctokitClient } from './util/constants';
+import type { RestEndpointMethodTypes } from '@octokit/rest';
+import { logger } from './util/debug';
+
+const debug = logger('isMemberOf');
+
+const ACTIVE_STATE = 'active';
+
+type GetMembershipForUserInOrg = RestEndpointMethodTypes['teams']['getMembershipForUserInOrg']['response'];
+
+const isGetMembershipForUserInOrg = (response: Record<string, unknown>): response is GetMembershipForUserInOrg => !!response?.data
+export const handleMembership = async (
+  client: OctokitClient,
+  isMemberOf: string[] = [],
+  requestConcurrency = 2
+): Promise<boolean> => {
+  const { repo, actor } = github.context;
+
+  const queue = new PQueue({ concurrency: requestConcurrency });
+  const membershipChecks = isMemberOf.map((team) => {
+    return {
+      org: repo.owner,
+      team_slug: team,
+      username: actor,
+    };
+  });
+  debug(
+    `We will check membership of ${actor} in the following teams ${membershipChecks.map(({ team_slug: team }) => team)}`
+  );
+
+  const results = await Promise.all(
+    membershipChecks.map((membership) => queue.add(() => client.teams.getMembershipForUserInOrg(membership)))
+  ).catch(catchHandler(debug));
+
+  return (results as Record<string, unknown>[]).some((response: Record<string, unknown>) => {
+    return isGetMembershipForUserInOrg(response) && response.data.state == ACTIVE_STATE;
+  });
```

Comment:
I think this bit of code is problematic, but you'll have to verify for me.

If I don't belong in one of the teams which we check for membership, then `catchHandler` will have to handle a `404` not found. This needs to be added to: https://github.com/gagoar/use-herald-action/blob/cb11e2c0b432984678a6d34514e9737a19badd5a/src/util/constants.ts#L52-L54

Furthermore, if any of the promises reject, I'm not sure that `results` is an array anymore, which is where I run into the error complaining that `results.some` is not a function. I think at that point, `results` is simply [the value of the promise which rejected](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all#rejection).

FYI even after I added `404` to the `AllowedHttpErrors` enum, I am still seeing the `results.some` is not a function error.

Reply:
mm, how is that not happening on the code?